neo4j/internal/bolt/bolt4.go

```diff
@@ -766,8 +763,12 @@ func (b *bolt4) receiveNext() (*db.Record, bool, *db.Summary) {
 		sum.ServerVersion = b.serverVersion
 		sum.ServerName = b.serverName
 		sum.TFirst = b.tfirst
+		if len(sum.Bookmark) > 0 {
```

Comment:
the previous version used to do this when a stream was closed (via `b.streams.onClose`)
At first glance, this seems to change the bookmark behavior.
It does because closing a session or switching to another stream forces the current stream consumption leading to this "success" code path here?

Reply:
There will only be a bookmark when hasMore is false so it shouldn't be an issue. Might be that we don't even need an if len(...) anymore. It was needed before because it was invoked in more situations. Good catch though!